extensions/html-language-features/server/src/modes/javascriptMode.ts

```diff
@@ -302,10 +319,12 @@ export function getJavaScriptMode(documentRegions: LanguageModelCache<HTMLDocume
 			const jsLanguageService = await host.getLanguageService(jsDocument);
 			const definition = jsLanguageService.getDefinitionAtPosition(jsDocument.uri, jsDocument.offsetAt(position));
 			if (definition) {
-				return definition.filter(d => d.fileName === jsDocument.uri).map(d => {
+				return definition.map(d => {
+					const text = jsLanguageService.getProgram()!.getSourceFile(d.fileName)!.text;
+					const doc = TextDocument.create('tmp', 'javascript', 1, text);
```

Comment:
Convert source files into `TextDocument` objects to do span conversion. (There may be a better way to do this).